.github/workflows/cron_weekend.yaml

```
@@ -0,0 +1,38 @@
+name: Weekend Cleanup
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * SAT'
+
+jobs:
+  fetchWebhooks:
+    runs-on: ubuntu-latest
+    outputs:
+      hook_ids: ${{ steps.format.outputs.hook_ids }}
+
+    steps:
+      - uses: octokit/request-action@v2.x
+        id: request
+        with:
+          route: GET /repos/hashicorp/test-policy-set/hooks
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+      - id: format
+        run: |
+          echo "hook_ids=$(jq -n '${{ steps.request.outputs.data }}' | jq -rc '[.[] .id]')\n" >> $GITHUB_OUTPUT
+      - run: |
+          echo 'Deleting these webhook ids: ${{ steps.format.outputs.hook_ids }}'
+  cleanWebhooks:
+    runs-on: ubuntu-latest
+    needs: [ fetchWebhooks ]
+    if: fromJSON(needs.fetchWebhooks.outputs.hook_ids)[0] != null
+    strategy:
+      matrix:
+        value: ${{fromJSON(needs.fetchWebhooks.outputs.hook_ids)}}
+    steps:
+      - uses: octokit/request-action@v2.x
+        id: delete_webhook
+        with:
+          route: DELETE /repos/hashicorp/test-policy-set/hooks/${{ matrix.value }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```

I modified the source repo to allow it to be called from any other repo. You could do it this way, too (I think)

Here's the docs https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

```suggestion
jobs:
  call-delete-all-webhooks:
    steps:
      - uses: hashicorp/terraform-random-1/.github/workflows/delete_all_webhooks.yml@main
        secrets:
          GH_TOKEN: ${{ secrets.GH_TOKEN }}
```

---

Ah yeah, I quite prefer that! I was thinking we would probably publish an action once we were happy with the way it works, but this is a nice lightweight way to dedupe for now.